actionview/lib/action_view/helpers/form_tag_helper.rb

```diff
@@ -78,6 +78,37 @@ def form_tag(url_for_options = {}, options = {}, &block)
         end
       end
 
+      # Generate an HTML <tt>id</tt> attribute value for the given name and
+      # field combination
+      #
+      # Return the value generated by the <tt>FormBuilder</tt> for the given
+      # attribute name.
+      #
+      #   <%= label_tag :post, :title %>
+      #   <%= text_field_tag :post, :title, aria: { describedby: field_id(:post, :title, :error) } %>
+      #   <span id="<%= field_id(:post, :title, :error) %>">is blank</span>
+      #
+      # In the example above, the <tt><input type="text"></tt> element built by
+      # the call to <tt>text_field_tag</tt> declares an
+      # <tt>aria-describedby</tt> attribute referencing the <tt><span></tt>
+      # element, sharing a common <tt>id</tt> root (<tt>post_title</tt>, in this
+      # case).
+      def field_id(object_name, name, *suffixes, index: nil)
```

Comment:
Copying https://github.com/rails/rails/pull/40127#discussion_r478799505 here, since the line in question has changed.

This set of changes was lifted directly from `ActionView::Helpers::Tags::Base`.

https://github.com/rails/rails/blob/7101489293ece071013417aeed2da3aa1b0927c9/actionview/lib/action_view/helpers/tags/base.rb#L119-L137

Given the "a little duplication to construct fewer strings", it seemed like that code was finely tuned for performance, which made me hesitant to abstract out a shared and re-usable chunk of code.

My gut tells me that `FormBuilder#id` will be invoked less frequently than the `Tags::Base` code. That being said, if there _is_ an abstraction that feels right, I'm all for extracting it.

Reply:
My first attempt is to extract out a shared constant (as introduced in https://github.com/rails/rails/pull/40127/commits/419e1792619f250ef396fa4dc004f0d707b30cd3).

Reply:
AFAIK `ActionView::Helpers::Tags::Base` has acces to the template so we should probably just move `tag_id` implementation to `field_id` and call `field_id` from `tag_id`.

Reply:
@rafaelfranca I'll try that out. The `# a little duplication to construct fewer strings` comments make me a little nervous about moving that code around. 

To move that definition elsewhere, we'd likely lose some of the memoization that's going on. Is there a way to gauge the impact of these changes?

Reply:
@rafaelfranca I've pushed up https://github.com/rails/rails/commit/c5869ce9cca414165f14d17165dfafe8d019b673 to try that out.

Reply:
Right, that is why I suggested moving the entire code, so we keep the `# a little duplication to construct fewer strings` comment and the duplication to construct fewer strings.